integrations/huggingface_hub/scripts/push_to_hub.py

```diff
@@ -0,0 +1,179 @@
+    precision = create_metric("Precision", "precision", precision)
+    recall = create_metric("Recall", "recall", recall)
+    f_score = create_metric("F Score", "f_score", f_score)
+
+    return [precision, recall, f_score]
+
+def create_model_index(repo_name, data):
+    model_index = {"name": repo_name}
+
+    results = []
+    if "ents_p" in data:
+        results.append({
+            "tasks": {
+                "name": "NER",
+                "type": "token-classification",
+                "metrics": create_p_r_f_list(data["ents_p"], data["ents_r"], data["ents_f"])
+            }
+        })
+    if "tag_acc" in data:
+        results.append({
+            "tasks": {
+                "name": "POS",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["tag_acc"])]
+            }
+        })
+    if "sents_p" in data:
+        results.append({
+            "tasks": {
+                "name": "SENTER",
+                "type": "token-classification",
+                "metrics": create_p_r_f_list(data["sents_p"], data["sents_r"], data["sents_f"])
+            }
+        })
+    if "dep_uas" in data:
+        results.append({
+            "tasks": {
+                "name": "UNLABELED_DEPENDENCIES",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["dep_uas"])]
+            }
+        })
+    if "dep_las" in data:
+        results.append({
+            "tasks": {
+                "name": "LABELED_DEPENDENCIES",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["dep_uas"])]
+            }
+        })
+
+    model_index["results"] = results
+    return model_index
+
+
+def create_model_card(repo_name, repo_dir):
+    with open(os.path.join(repo_dir, "meta.json")) as f:
+        data = json.load(f)
+        lang = data["lang"] if data["lang"] != "xx" else "multilingual"
+        lic = data["license"].replace(" ", "-")
+        tags = ["spacy"]
+        for component in data["components"]:
+            if component in token_classification_components and "token-classification" not in tags:
+                tags.append("token-classification")
+            if component in text_classification_components and "text-classification" not in tags:
+                tags.append("text-classification")
+
+        metadata = _insert_values_as_list({}, "tags", tags)
+        metadata = _insert_values_as_list(metadata, "language", lang)
+        metadata = _insert_value(metadata, "license", lic)
+        metadata["model-index"]  = create_model_index(repo_name, data["performance"])
+        metadata = yaml.dump(metadata, sort_keys=False)
+
+
+        metadata_section = f"---\n{metadata}---\n"
+
+        # Read README generated by package
+        readme_path = os.path.join(repo_dir, "README.md")
+        readme = ""
+        with open(readme_path) as f:
```

Comment:
We might as well do a quick `exists` check here in case a user removed their `README.md` (they shouldn't, but I guess they should be allowed to 😛)

Reply:
Done!